doc/vhpidirect/examples/quickstart.rst

```diff
@@ -61,23 +63,35 @@ Because logic values are more than high and low (``1`` and ``0``), the VHDL valu
 7. 'H' 
 8. '-' 
 
-Because number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C).
+Because number of enumeration values is less than 256, logic values are transported in 8 bit words (a ``char`` type in C) (as mentioned in :ref:`Restrictions_on_foreign_declarations`).
 Logic vectors, of a bounded size, are easily created in C and passed to VHDL as ``char[]``. VHDL receives pointers as ``access`` types, in this case an access of a subtype of std_logic_vector.
 
-Providing logic values in C as their enumeration numbers is simplified with the use of a matching enumeration, provided as ``HDL_LOGIC_STATES``. Printing out a logic value's associated character is also simplified with the provided ``const char HDL_LOGIC_CHAR[]``.
+In this example two declarations make handling logic values in C a bit easier:
+
+- Providing logic values in C as their enumeration numbers is simplified with the use of a matching enumeration, ``HDL_LOGIC_STATES``. 
+- Printing out a logic value's associated character is also simplified with the ``const char HDL_LOGIC_CHAR[]`` declaration.
+
+This example builds on the matrices example (:ref:`COSIM:VHPIDIRECT:Examples:quickstart:matrices:`). Foreign subprograms are declared that enable receiving the size of two different logic vectors as well as the vectors themselves from C. There is only one subprogram to get the size of both C arrays, and it takes in an integer to determine which array's size gets returned. 
 
-This example declares foreign subprograms that enable receiving the size of two different logic vectors as well as the vectors themselves from C. The two vectors are populated with logic values in different ways:
+.. HINT::
+  The ``getLogicVecSize`` in VHDL is declared as receiving a ``boolean`` argument. In C the function is declared to receive an ``integer`` argument. The VHDL booleans ``true`` and ``false`` have integer values, ``1`` and ``0`` respectively. So this mismatch causes no real issues.
```

Comment:
Is there any reference that explain why booleans are mapped as integers and not uint8_t or int64_t?

Reply:
I think that the bits in int types are the same between unsigned and signed for the positive values. And the different widths just include more significant bits. [0, 1] never cause any of these differences to come into effect. 

I don't recall any mentions of booleans within ghdl VHPIDIRECT no...

Reply:
> I think that the bits in int types are the same between unsigned and signed for the positive values. And the different widths just include more significant bits. [0, 1] never cause any of these differences to come into effect.

While this is true in VHDL, it might not be so in C. If VHDL saves a boolean as a byte, you get a pointer and the dereference it as an integer in C, you will read 3 bytes which you should not. In most cases, those will be empty/zero, because memory allocation is aligned. But it might lead to annoying bugs.

I asked Tristan, just in case, and he told me that `It's 0 or 1 stored on a byte (a char)`. Hence, we should both fix it and add it to the section about declarations/restrictions of the docs.

Reply:
I'd like to add the mention to the docs in a separate PR after this is merged. I will reference the section in this example's docs too.